src/main/groovy/com/brightsparklabs/gradle/baseline/BaselinePluginExtension.groovy

```diff
@@ -47,6 +47,25 @@ class S3DeployConfig {
     /** [Optional] The prefix to prepend to uploaded files. Default: "". */
     String prefix = ""
 
-    /** The paths of the files to upload to the S3 bucket. */
+    /**
+     * The absolute filepaths of the files to upload to the S3 bucket. Each filepath is treated
+     * as a regex, and matched against all files in the project's `build` directory. All matched
+     * files are uploaded.
+     */
     Set<String> filesToUpload
-}
\ No newline at end of file
+
+    /**
+     * [Optional] The endpoint to upload files to. This value overrides the default AWS
+     * endpoint, and allows files to be uploaded to any S3-compatible storage. For example, files
+     * could be uploaded to a local MinIO instance by setting this value to
+     * "http://localhost:9000". Default: unset.
+     */
+    String endpointOverride
```

Comment:
Can we actually make this `Optional<String>`? Not sure if that works in extension classes. But if it does, it cleans up stuff above:

```
                profile.IfPresent(p -> {
                    final ClientOverrideConfiguration.Builder s3OverrideConfigBuilder =
                            ClientOverrideConfiguration.builder()
                    s3OverrideConfigBuilder.defaultProfileName(p)
                    final s3OverrideConfig = s3OverrideConfigBuilder.build()
                    s3Builder.overrideConfiguration(s3OverrideConfig)
                }
```

Reply:
This throws an error when given a String as input:
```
Cannot cast object 'http://localhost:9000' with class 'java.lang.String' to class 'java.util.Optional'
```
I found an intermediate solution by casting the input to `Optional<String>` within the setup function.